手动上传的主题在线更新失败 repository not found

### 是什么版本出现了此问题？

1.4.11

### 使用的什么数据库？

H2

### 使用的哪种方式部署？

Fat Jar

### 在线站点地址

_No response_

### 发生了什么？

首次初始化 Halo，更新默认主题提示失败。

### 相关日志输出

```shell
run.halo.app.exception.ThemeUpdateException: 主题更新失败！
	at run.halo.app.service.impl.ThemeServiceImpl.update(ThemeServiceImpl.java:375) ~[classes!/:1.4.11]
	at run.halo.app.service.impl.ThemeServiceImpl$$FastClassBySpringCGLIB$$aa446525.invoke(<generated>) ~[classes!/:1.4.11]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at run.halo.app.service.impl.ThemeServiceImpl$$EnhancerBySpringCGLIB$$4ce30b8b.update(<generated>) ~[classes!/:1.4.11]
	at run.halo.app.controller.admin.api.ThemeController.updateThemeByFetching(ThemeController.java:196) ~[classes!/:1.4.11]
	at run.halo.app.controller.admin.api.ThemeController$$FastClassBySpringCGLIB$$e8e9d45c.invoke(<generated>) ~[classes!/:1.4.11]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at run.halo.app.core.ControllerLogAop.controller(ControllerLogAop.java:48) ~[classes!/:1.4.11]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692) ~[spring-aop-5.3.8.jar!/:5.3.8]
	at run.halo.app.controller.admin.api.ThemeController$$EnhancerBySpringCGLIB$$2aff8161.updateThemeByFetching(<generated>) ~[classes!/:1.4.11]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1063) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at org.springframework.web.servlet.FrameworkServlet.doPut(FrameworkServlet.java:920) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:520) ~[jakarta.servlet-api-4.0.4.jar!/:4.0.4]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883) ~[spring-webmvc-5.3.8.jar!/:5.3.8]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:584) ~[jakarta.servlet-api-4.0.4.jar!/:4.0.4]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:228) ~[websocket-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at run.halo.app.security.filter.AdminAuthenticationFilter.doAuthenticate(AdminAuthenticationFilter.java:120) ~[classes!/:1.4.11]
	at run.halo.app.security.filter.AbstractAuthenticationFilter.doFilterInternal(AbstractAuthenticationFilter.java:229) ~[classes!/:1.4.11]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:103) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:103) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at run.halo.app.filter.CorsFilter.doFilter(CorsFilter.java:53) ~[classes!/:1.4.11]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96) ~[spring-boot-actuator-2.5.1.jar!/:2.5.1]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at run.halo.app.filter.LogFilter.doFilterInternal(LogFilter.java:40) ~[classes!/:1.4.11]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.8.jar!/:5.3.8]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) ~[jetty-security-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501) ~[jetty-servlet-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) ~[jetty-server-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) ~[jetty-io-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) ~[jetty-io-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) ~[jetty-io-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036) ~[jetty-util-9.4.42.v20210604.jar!/:9.4.42.v20210604]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]
Caused by: org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: /home/ryanwang/.halo/templates/themes/anatole
	at org.eclipse.jgit.lib.BaseRepositoryBuilder.build(BaseRepositoryBuilder.java:585) ~[org.eclipse.jgit-5.9.0.202009080501-r.jar!/:5.9.0.202009080501-r]
	at org.eclipse.jgit.api.Git.open(Git.java:93) ~[org.eclipse.jgit-5.9.0.202009080501-r.jar!/:5.9.0.202009080501-r]
	at org.eclipse.jgit.api.Git.open(Git.java:73) ~[org.eclipse.jgit-5.9.0.202009080501-r.jar!/:5.9.0.202009080501-r]
	at run.halo.app.theme.GitThemeUpdater.merge(GitThemeUpdater.java:79) ~[classes!/:1.4.11]
	at run.halo.app.theme.GitThemeUpdater.update(GitThemeUpdater.java:54) ~[classes!/:1.4.11]
	at run.halo.app.service.impl.ThemeServiceImpl.update(ThemeServiceImpl.java:370) ~[classes!/:1.4.11]
	... 113 common frames omitted
```


### 附加信息

_No response_

--- a/src/main/java/run/halo/app/core/CommonResultControllerAdvice.java
+++ b/src/main/java/run/halo/app/core/CommonResultControllerAdvice.java
@@ -1,6 +1,5 @@
 package run.halo.app.core;
 
-import org.jetbrains.annotations.NotNull;
 import org.springframework.core.MethodParameter;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
@@ -26,18 +25,18 @@ public class CommonResultControllerAdvice implements ResponseBodyAdvice<Object> 
 
     @Override
     public boolean supports(MethodParameter returnType,
-        @NotNull Class<? extends HttpMessageConverter<?>> converterType) {
+        @NonNull Class<? extends HttpMessageConverter<?>> converterType) {
         return AbstractJackson2HttpMessageConverter.class.isAssignableFrom(converterType);
     }
 
     @Override
     @NonNull
     public final Object beforeBodyWrite(@Nullable Object body,
-        @NotNull MethodParameter returnType,
-        @NotNull MediaType contentType,
-        @NotNull Class<? extends HttpMessageConverter<?>> converterType,
-        @NotNull ServerHttpRequest request,
-        @NotNull ServerHttpResponse response) {
+        @NonNull MethodParameter returnType,
+        @NonNull MediaType contentType,
+        @NonNull Class<? extends HttpMessageConverter<?>> converterType,
+        @NonNull ServerHttpRequest request,
+        @NonNull ServerHttpResponse response) {
         MappingJacksonValue container = getOrCreateContainer(body);
         // The contain body will never be null
         beforeBodyWriteInternal(container, contentType, returnType, request, response);

--- a/src/main/java/run/halo/app/listener/StartedListener.java
+++ b/src/main/java/run/halo/app/listener/StartedListener.java
@@ -167,12 +167,9 @@ public class StartedListener implements ApplicationListener<ApplicationStartedEv
             // Create theme folder
             Path themePath = themeService.getBasePath();
 
-            // Fix the problem that the project cannot start after moving to a new server
-            if (Files.notExists(themePath) || !isInstalled) {
+            if (themeService.fetchThemePropertyBy(HaloConst.DEFAULT_THEME_ID).isEmpty()) {
                 FileUtils.copyFolder(source, themePath);
-                log.debug("Copied theme folder from [{}] to [{}]", source, themePath);
-            } else {
-                log.debug("Skipped copying theme folder due to existence of theme folder");
+                log.info("Copied theme folder from [{}] to [{}]", source, themePath);
             }
         } catch (Exception e) {
             if (e instanceof FileNotFoundException) {

--- a/src/main/java/run/halo/app/repository/ThemeRepositoryImpl.java
+++ b/src/main/java/run/halo/app/repository/ThemeRepositoryImpl.java
@@ -98,10 +98,6 @@ public class ThemeRepositoryImpl
 
     @Override
     public Optional<ThemeProperty> fetchThemePropertyByThemeId(String themeId) {
-        if (StringUtils.equals(themeId, getActivatedThemeId())) {
-            return Optional.of(getActivatedThemeProperty());
-        }
-
         return ThemePropertyScanner.INSTANCE.scan(getThemeRootPath(), null)
             .stream()
             .filter(property -> Objects.equals(themeId, property.getId()))

--- a/src/main/java/run/halo/app/service/impl/ThemeServiceImpl.java
+++ b/src/main/java/run/halo/app/service/impl/ThemeServiceImpl.java
@@ -27,11 +27,13 @@ import run.halo.app.exception.BadRequestException;
 import run.halo.app.exception.ForbiddenException;
 import run.halo.app.exception.NotFoundException;
 import run.halo.app.exception.ServiceException;
+import run.halo.app.exception.ThemeNotFoundException;
 import run.halo.app.exception.ThemeNotSupportException;
 import run.halo.app.exception.ThemePropertyMissingException;
 import run.halo.app.exception.ThemeUpdateException;
 import run.halo.app.handler.theme.config.ThemeConfigResolver;
 import run.halo.app.handler.theme.config.support.Group;
+import run.halo.app.handler.theme.config.support.Option;
 import run.halo.app.handler.theme.config.support.ThemeProperty;
 import run.halo.app.model.support.HaloConst;
 import run.halo.app.model.support.ThemeFile;
@@ -131,7 +133,7 @@ public class ThemeServiceImpl implements ThemeService {
             Path themePath = Paths.get(themeProperty.getThemePath());
             try (Stream<Path> pathStream = Files.list(themePath)) {
                 return pathStream.filter(path ->
-                    StringUtils.startsWithIgnoreCase(path.getFileName().toString(), prefix))
+                        StringUtils.startsWithIgnoreCase(path.getFileName().toString(), prefix))
                     .map(path -> {
                         // Remove prefix
                         final var customTemplate = StringUtils
@@ -315,13 +317,17 @@ public class ThemeServiceImpl implements ThemeService {
     @Override
     @NonNull
     public ThemeProperty getActivatedTheme() {
-        return fetchActivatedTheme().orElseThrow();
+        return themeRepository.getActivatedThemeProperty();
     }
 
     @Override
     @NonNull
     public Optional<ThemeProperty> fetchActivatedTheme() {
-        return Optional.of(themeRepository.getActivatedThemeProperty());
+        try {
+            return Optional.of(themeRepository.getActivatedThemeProperty());
+        } catch (ThemeNotFoundException tne) {
+            return Optional.empty();
+        }
     }
 
     @Override

--- a/src/main/java/run/halo/app/theme/GitThemeUpdater.java
+++ b/src/main/java/run/halo/app/theme/GitThemeUpdater.java
@@ -76,7 +76,7 @@ public class GitThemeUpdater implements ThemeUpdater {
 
         final var oldThemePath = Paths.get(oldThemeProperty.getThemePath());
         // open old git repo
-        try (final var oldGit = Git.open(oldThemePath.toFile())) {
+        try (final var oldGit = Git.init().setDirectory(oldThemePath.toFile()).call()) {
             // 0. commit old repo
             commitAutomatically(oldGit);
 

--- a/src/main/java/run/halo/app/utils/footnotes/Footnote.java
+++ b/src/main/java/run/halo/app/utils/footnotes/Footnote.java
@@ -7,7 +7,7 @@ import com.vladsch.flexmark.util.ast.Document;
 import com.vladsch.flexmark.util.ast.Node;
 import com.vladsch.flexmark.util.ast.ReferencingNode;
 import com.vladsch.flexmark.util.sequence.BasedSequence;
-import org.jetbrains.annotations.NotNull;
+import org.springframework.lang.NonNull;
 import run.halo.app.utils.footnotes.internal.FootnoteRepository;
 
 /**
@@ -30,7 +30,7 @@ public class Footnote extends Node implements DelimitedNode, DoNotDecorate, Link
 
     protected int referenceOrdinal;
 
-    @NotNull
+    @NonNull
     @Override
     public BasedSequence getReference() {
         return text;
@@ -80,14 +80,14 @@ public class Footnote extends Node implements DelimitedNode, DoNotDecorate, Link
         this.footnoteBlock = footnoteBlock;
     }
 
-    @NotNull
+    @NonNull
     @Override
     public BasedSequence[] getSegments() {
         return new BasedSequence[] {openingMarker, text, closingMarker};
     }
 
     @Override
-    public void getAstExtra(@NotNull StringBuilder out) {
+    public void getAstExtra(@NonNull StringBuilder out) {
         out.append(" ordinal: ")
             .append(footnoteBlock != null ? footnoteBlock.getFootnoteOrdinal() : 0).append(" ");
         delimitedSegmentSpanChars(out, openingMarker, text, closingMarker, "text");

--- a/src/main/java/run/halo/app/utils/footnotes/FootnoteBlock.java
+++ b/src/main/java/run/halo/app/utils/footnotes/FootnoteBlock.java
@@ -10,8 +10,8 @@ import com.vladsch.flexmark.util.data.DataHolder;
 import com.vladsch.flexmark.util.sequence.BasedSequence;
 import com.vladsch.flexmark.util.sequence.SequenceUtils;
 import java.util.Objects;
-import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
 import run.halo.app.utils.footnotes.internal.FootnoteRepository;
 
 /**
@@ -43,7 +43,7 @@ public class FootnoteBlock extends Block
 
     @Nullable
     @Override
-    public Footnote getReferencingNode(@NotNull Node node) {
+    public Footnote getReferencingNode(@NonNull Node node) {
         return node instanceof Footnote ? (Footnote) node : null;
     }
 
@@ -74,7 +74,7 @@ public class FootnoteBlock extends Block
     }
 
     @Override
-    public void getAstExtra(@NotNull StringBuilder out) {
+    public void getAstExtra(@NonNull StringBuilder out) {
         out.append(" ordinal: ").append(footnoteOrdinal).append(" ");
         segmentSpan(out, openingMarker, "open");
         segmentSpan(out, text, "text");
@@ -82,7 +82,7 @@ public class FootnoteBlock extends Block
         segmentSpan(out, footnote, "footnote");
     }
 
-    @NotNull
+    @NonNull
     @Override
     public BasedSequence[] getSegments() {
         return new BasedSequence[] {openingMarker, text, closingMarker, footnote};

--- a/src/main/java/run/halo/app/utils/footnotes/FootnoteExtension.java
+++ b/src/main/java/run/halo/app/utils/footnotes/FootnoteExtension.java
@@ -9,7 +9,7 @@ import com.vladsch.flexmark.util.data.DataKey;
 import com.vladsch.flexmark.util.data.MutableDataHolder;
 import com.vladsch.flexmark.util.format.options.ElementPlacement;
 import com.vladsch.flexmark.util.format.options.ElementPlacementSort;
-import org.jetbrains.annotations.NotNull;
+import org.springframework.lang.NonNull;
 import run.halo.app.utils.footnotes.internal.FootnoteBlockParser;
 import run.halo.app.utils.footnotes.internal.FootnoteLinkRefProcessor;
 import run.halo.app.utils.footnotes.internal.FootnoteNodeFormatter;
@@ -63,8 +63,8 @@ public class FootnoteExtension
     }
 
     @Override
-    public void extend(@NotNull HtmlRenderer.Builder htmlRendererBuilder,
-        @NotNull String rendererType) {
+    public void extend(@NonNull HtmlRenderer.Builder htmlRendererBuilder,
+        @NonNull String rendererType) {
         if (htmlRendererBuilder.isRendererType("HTML")) {
             htmlRendererBuilder.nodeRendererFactory(new FootnoteNodeRenderer.Factory());
         }
@@ -77,7 +77,7 @@ public class FootnoteExtension
     }
 
     @Override
-    public void rendererOptions(@NotNull MutableDataHolder options) {
+    public void rendererOptions(@NonNull MutableDataHolder options) {
 
     }
 

--- a/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteBlockParser.java
+++ b/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteBlockParser.java
@@ -16,8 +16,8 @@ import com.vladsch.flexmark.util.sequence.BasedSequence;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
 import run.halo.app.utils.footnotes.FootnoteBlock;
 import run.halo.app.utils.footnotes.FootnoteExtension;
 
@@ -115,9 +115,9 @@ public class FootnoteBlockParser extends AbstractBlockParser {
             return false;
         }
 
-        @NotNull
+        @NonNull
         @Override
-        public BlockParserFactory apply(@NotNull DataHolder options) {
+        public BlockParserFactory apply(@NonNull DataHolder options) {
             return new BlockFactory(options);
         }
     }

--- a/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteLinkRefProcessor.java
+++ b/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteLinkRefProcessor.java
@@ -6,7 +6,7 @@ import com.vladsch.flexmark.util.ast.Document;
 import com.vladsch.flexmark.util.ast.Node;
 import com.vladsch.flexmark.util.data.DataHolder;
 import com.vladsch.flexmark.util.sequence.BasedSequence;
-import org.jetbrains.annotations.NotNull;
+import org.springframework.lang.NonNull;
 import run.halo.app.utils.footnotes.Footnote;
 import run.halo.app.utils.footnotes.FootnoteBlock;
 import run.halo.app.utils.footnotes.FootnoteExtension;
@@ -33,14 +33,14 @@ public class FootnoteLinkRefProcessor implements LinkRefProcessor {
     }
 
     @Override
-    public boolean isMatch(@NotNull BasedSequence nodeChars) {
+    public boolean isMatch(@NonNull BasedSequence nodeChars) {
         return nodeChars.length() >= 3 && nodeChars.charAt(0) == '[' && nodeChars.charAt(1) == '^'
             && nodeChars.endCharAt(1) == ']';
     }
 
-    @NotNull
+    @NonNull
     @Override
-    public Node createNode(@NotNull BasedSequence nodeChars) {
+    public Node createNode(@NonNull BasedSequence nodeChars) {
         BasedSequence footnoteId = nodeChars.midSequence(2, -1).trim();
         FootnoteBlock footnoteBlock =
             footnoteId.length() > 0 ? footnoteRepository.get(footnoteId.toString()) : null;
@@ -55,39 +55,39 @@ public class FootnoteLinkRefProcessor implements LinkRefProcessor {
         return footnote;
     }
 
-    @NotNull
+    @NonNull
     @Override
-    public BasedSequence adjustInlineText(@NotNull Document document, @NotNull Node node) {
+    public BasedSequence adjustInlineText(@NonNull Document document, @NonNull Node node) {
         assert node instanceof Footnote;
         return ((Footnote) node).getText();
     }
 
     @Override
-    public boolean allowDelimiters(@NotNull BasedSequence chars, @NotNull Document document,
-        @NotNull Node node) {
+    public boolean allowDelimiters(@NonNull BasedSequence chars, @NonNull Document document,
+        @NonNull Node node) {
         return true;
     }
 
     @Override
-    public void updateNodeElements(@NotNull Document document, @NotNull Node node) {
+    public void updateNodeElements(@NonNull Document document, @NonNull Node node) {
 
     }
 
     public static class Factory implements LinkRefProcessorFactory {
 
-        @NotNull
+        @NonNull
         @Override
-        public LinkRefProcessor apply(@NotNull Document document) {
+        public LinkRefProcessor apply(@NonNull Document document) {
             return new FootnoteLinkRefProcessor(document);
         }
 
         @Override
-        public boolean getWantExclamationPrefix(@NotNull DataHolder options) {
+        public boolean getWantExclamationPrefix(@NonNull DataHolder options) {
             return WANT_EXCLAMATION_PREFIX;
         }
 
         @Override
-        public int getBracketNestingLevel(@NotNull DataHolder options) {
+        public int getBracketNestingLevel(@NonNull DataHolder options) {
             return BRACKET_NESTING_LEVEL;
         }
     }

--- a/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteNodeFormatter.java
+++ b/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteNodeFormatter.java
@@ -15,8 +15,8 @@ import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
-import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
 import run.halo.app.utils.footnotes.Footnote;
 import run.halo.app.utils.footnotes.FootnoteBlock;
 import run.halo.app.utils.footnotes.FootnoteExtension;
@@ -100,9 +100,9 @@ public class FootnoteNodeFormatter
 
     public static class Factory implements NodeFormatterFactory {
 
-        @NotNull
+        @NonNull
         @Override
-        public NodeFormatter create(@NotNull DataHolder options) {
+        public NodeFormatter create(@NonNull DataHolder options) {
             return new FootnoteNodeFormatter(options);
         }
     }

--- a/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteNodeRenderer.java
+++ b/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteNodeRenderer.java
@@ -18,7 +18,7 @@ import java.util.HashSet;
 import java.util.Locale;
 import java.util.Set;
 import org.apache.commons.lang3.StringUtils;
-import org.jetbrains.annotations.NotNull;
+import org.springframework.lang.NonNull;
 import run.halo.app.utils.footnotes.Footnote;
 import run.halo.app.utils.footnotes.FootnoteBlock;
 import run.halo.app.utils.footnotes.FootnoteExtension;
@@ -53,8 +53,8 @@ public class FootnoteNodeRenderer implements PhasedNodeRenderer {
     }
 
     @Override
-    public void renderDocument(@NotNull NodeRendererContext context, @NotNull HtmlWriter html,
-        @NotNull Document document, @NotNull RenderingPhase phase) {
+    public void renderDocument(@NonNull NodeRendererContext context, @NonNull HtmlWriter html,
+        @NonNull Document document, @NonNull RenderingPhase phase) {
         if (phase == RenderingPhase.BODY_TOP) {
             if (recheckUndefinedReferences) {
                 // need to see if have undefined footnotes that were defined after parsing
@@ -171,9 +171,9 @@ public class FootnoteNodeRenderer implements PhasedNodeRenderer {
 
     public static class Factory implements NodeRendererFactory {
 
-        @NotNull
+        @NonNull
         @Override
-        public NodeRenderer apply(@NotNull DataHolder options) {
+        public NodeRenderer apply(@NonNull DataHolder options) {
             return new FootnoteNodeRenderer(options);
         }
     }

--- a/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteRepository.java
+++ b/src/main/java/run/halo/app/utils/footnotes/internal/FootnoteRepository.java
@@ -13,7 +13,7 @@ import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import org.jetbrains.annotations.NotNull;
+import org.springframework.lang.NonNull;
 import run.halo.app.utils.footnotes.Footnote;
 import run.halo.app.utils.footnotes.FootnoteBlock;
 import run.halo.app.utils.footnotes.FootnoteExtension;
@@ -79,19 +79,19 @@ public class FootnoteRepository extends NodeRepository<FootnoteBlock> {
         super(FootnoteExtension.FOOTNOTES_KEEP.get(options));
     }
 
-    @NotNull
+    @NonNull
     @Override
     public DataKey<FootnoteRepository> getDataKey() {
         return FootnoteExtension.FOOTNOTES;
     }
 
-    @NotNull
+    @NonNull
     @Override
     public DataKey<KeepType> getKeepDataKey() {
         return FootnoteExtension.FOOTNOTES_KEEP;
     }
 
-    @NotNull
+    @NonNull
     @Override
     public Set<FootnoteBlock> getReferencedElements(Node parent) {
         HashSet<FootnoteBlock> references = new HashSet<>();

